samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -321,11 +325,177 @@ static struct bt_mesh_sensor time_since_presence_detected = {
 	.get = time_since_presence_detected_get,
 };
 
+static void amb_light_level_gain_get(struct bt_mesh_sensor_srv *srv,
+				     struct bt_mesh_sensor *sensor,
+				     const struct bt_mesh_sensor_setting *setting,
+				     struct bt_mesh_msg_ctx *ctx,
+				     struct sensor_value *rsp)
+{
+	(void)sensor_value_from_double(rsp, amb_light_level_gain);
+	printk("Ambient light level gain: %f\n", amb_light_level_gain);
+};
+
+static void amb_light_level_gain_store(double gain)
+{
+	amb_light_level_gain = gain;
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		int err;
+
+		err = settings_save_one("amb_light_level/gain",
+					&amb_light_level_gain, sizeof(amb_light_level_gain));
+		if (err) {
+			printk("Error storing setting (%d)\n", err);
+		} else {
+			printk("Stored setting\n");
+		}
+	}
+}
+
+static int amb_light_level_gain_set(struct bt_mesh_sensor_srv *srv,
+				    struct bt_mesh_sensor *sensor,
+				    const struct bt_mesh_sensor_setting *setting,
+				    struct bt_mesh_msg_ctx *ctx,
+				    const struct sensor_value *value)
+{
+	amb_light_level_gain_store(sensor_value_to_double(value));
+	printk("Ambient light level gain: %f\n", amb_light_level_gain);
+
+	return 0;
+}
+
+static int amb_light_level_ref_set(struct bt_mesh_sensor_srv *srv,
+				   struct bt_mesh_sensor *sensor,
+				   const struct bt_mesh_sensor_setting *setting,
+				   struct bt_mesh_msg_ctx *ctx,
+				   const struct sensor_value *value)
+{
+	double amb_light_level_ref = sensor_value_to_double(value);
+
+	/* When using the a real ambient light sensor the sensor value should be
+	 * read and used instead of the dummy value.
+	 */
+	amb_light_level_gain_store(amb_light_level_ref / dummy_ambient_light_value);
```

Comment:
Please do zero check on `dummy_ambient_light_value` to avoid divide by zero hard fault.